src/NUnitFramework/framework/List.cs

```diff
@@ -21,25 +21,30 @@
 // WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 // ***********************************************************************
 
+using System;
 using System.Collections;
 
 namespace NUnit.Framework
 {
-	/// <summary>
-	/// The List class is a helper class with properties and methods
-	/// that supply a number of constraints used with lists and collections.
-	/// </summary>
-	public class List
-	{
-		/// <summary>
-		/// List.Map returns a ListMapper, which can be used to map
-		/// the original collection to another collection.
-		/// </summary>
-		/// <param name="actual"></param>
-		/// <returns></returns>
-		public static ListMapper Map( ICollection actual )
-		{
-			return new ListMapper( actual );
-		}
-	}
+    /// <summary>
+    /// The List class is a helper class with properties and methods
+    /// that supply a number of constraints used with lists and collections.
+    /// </summary>
+    [Obsolete("The ListMapper class has been deprecated and will be removed in a future release. "
+        + "Please use the extension method System.Linq.Enumerable.Select instead.")]
+    public class List
+    {
+        /// <summary>
+        /// List.Map returns a ListMapper, which can be used to map
+        /// the original collection to another collection.
+        /// </summary>
+        /// <param name="actual"></param>
+        /// <returns></returns>
+        [Obsolete("The ListMapper class has been deprecated and will be removed in a future release. "
```

Comment:
If the class is obsolete, is it necessary to mark each method as well? This applies in several other places as well.

Reply:
This is a good point. We don't need duplicate warnings showing up.

The only way it would be useful to mark the method is if a third-party library does this:

```cs
public class ListEx : NUnit.Framework.List
{
}
```

If someone using that library does `ListEx.Map`, they wouldn't see an obsoletion warning. Seems far-fetched, will remove.

Reply:
Or, if the third-party library does this:

```cs
public class Foo
{
    public List List { get; }
}
```

And someone using the library does `new Foo(...).List.Map("foo")`. Still seems unlikely.